src/utils/embedded/embedded.provider.tsx

```diff
@@ -1078,6 +1083,52 @@ export function EmbeddedProvider({ children }: EmbeddedProviderProps) {
 
     lastUserIdRef.current = userId;
 
+    if (userId && userId !== await PersistentStorage.get<string>(StorageKeys.CONNECT.AUTH.USER_ID)) {
+      // TODO: This is a TEMP FIX to prevent users who share the same device from seeing someone else's tokens and
```

Comment:
@7i7o See here. However, it doesn't seem to work properly, as I only see AO despite calling `loadTokens()` and `updateAoToken()`. Once the page is reloaded, all tokens load fine. Any idea why this might happen?

Reply:
I see the tokens were updated to localStorage. When reading the tokens without hooks works but with hooks we don't see the actual tokens. Maybe issue with useStorage hook for Wander Connect usage with localStorage.